### PR TITLE
Tests - Alternative

### PR DESCRIPTION
**Pull request by mottosso**
Getting Travis to play nice.

- Alternative to https://github.com/getavalon/docker/pull/17

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+dist: trusty
+
+python:
+- 3.4
+
+services:
+  - docker
+
+script:
+  - docker build -t avalon .
```
Comment (tokejepsen) [inline .travis.yml]: Building to ```avalon``, but running ```docker```?

File: tests/test_cli.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import os
+import subprocess
+
+
+def test_backup_restore():
```
Comment (tokejepsen) [inline tests/test_cli.py]: There is a better version of this test:

```python
def test_backup_restore():
    """Backup and restore works"""

    # Populate database
    subprocess.call(["avalon", "--import", "batman"], shell=True)

    # Backup
    result = subprocess.call(["avalon", "--backup"], shell=True)
    assert result == 0

    # Get backup
    backup_path = ""
    for f in os.listdir(os.getcwd()):
        if f.endswith(".zip"):
            backup_path = os.path.join(os.getcwd(), f)

    # Wipe database
    client = pymongo.MongoClient(os.environ["AVALON_MONGO"])
    client.drop_database("avalon")

    # Restore database
    result = subprocess.call(["avalon", "--restore", backup_path], shell=True)
    assert result == 0

    # Clean up
    os.remove(backup_path)
```

File: run_tests.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import nose
+
+
+root = os.path.abspath(os.path.join(__file__, ".."))
+
+# Setup environment for executable
+os.environ["PATH"] += os.pathsep + os.path.join(root, "volume")
+
+# Build docker file
+subprocess.call(["docker", "build", ".", "-t", "getavalon/docker"], cwd=root)
```
Comment (tokejepsen) [inline run_tests.py]: Maybe the instructions to run the tests, needs to include the building step so running the tests do not actually build anything?

**Comment by mottosso** [comment]
Yes, as I pushed this I realised how long it'd take before seeing whether it worked or not. So I think I'll skip this and implement the separate containers instead.

That way we can skip building e.g. CGWire when only testing Mongo.

I'll try and get another PR up in the next hour or so.

**Pull request closed without merging**